walk/walk.go

```diff
@@ -165,28 +156,38 @@ func visit(c *config.Config, cexts []config.Configurer, knownDirectives map[stri
 		return
 	}
 
-	var subdirs, regularFiles []string
-	for _, ent := range ents {
+	// Filter and collect files
+	var regularFiles []string
+	for _, ent := range trie.files {
 		base := ent.Name()
 		entRel := path.Join(rel, base)
 		if wc.isExcluded(entRel) {
 			continue
 		}
-		ent := resolveFileInfo(wc, dir, entRel, ent)
-		switch {
-		case ent == nil:
+		if ent := resolveFileInfo(wc, dir, entRel, ent); ent != nil {
+			regularFiles = append(regularFiles, base)
+		}
+	}
+
+	// Filter and collect subdirectories
+	var subdirs []string
```

Comment:
Do we need this separate slice? I think that it just contains `t.entry.Name()` for ever subtrie.

Reply:
This array gets passed into the `WalkFunc` API.

Although we can move construction of this into the `if updateRels.shouldCall` block. Or can merge this loop and the next so we only have to construct 1/2 of these arrays...

Reply:
See the fixup commit